### PR TITLE
Pack traversal module

### PR DESCRIPTION
**Pull request by msimberg**
Includes `pack_traversal` and `unwrap`. Technically `pack_traversal` would not need to be together with `unwrap`. The latter depends on the former. Since they're practically only used together I put them in the same module. If someone thinks they should be separate I can do that as well.

### CODE REVIEW

File: libs/pack_traversal/src/pack_traversal.cpp [hunk 1]
```diff
@@ -5,30 +5,30 @@
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
```
Comment (hkaiser) [inline libs/pack_traversal/src/pack_traversal.cpp:5]: Shouldn't this file go into `.../tests/unit`?

File: libs/performance_counters/include/hpx/performance_counters/performance_counter_set.hpp [hunk 2]
```diff
@@ -12,10 +12,10 @@
 #include <hpx/errors.hpp>
 #include <hpx/lcos/dataflow.hpp>
 #include <hpx/lcos/future.hpp>
+#include <hpx/pack_traversal/unwrap.hpp>
```
Comment (hkaiser) [inline libs/performance_counters/include/hpx/performance_counters/performance_counter_set.hpp:15]: This module now depends on pack_traversal

File: libs/resource_partitioner/examples/async_customization.cpp [hunk 3]
```diff
@@ -26,8 +26,8 @@
 #include <hpx/functional/result_of.hpp>
 #include <hpx/lcos/dataflow.hpp>
 #include <hpx/lcos/when_all.hpp>
+#include <hpx/pack_traversal/pack_traversal.hpp>
```
Comment (hkaiser) [inline libs/resource_partitioner/examples/async_customization.cpp:29]: This module now depends on pack_traversal

File: libs/segmented_algorithms/include/hpx/parallel/segmented_algorithms/detail/scan.hpp [hunk 4]
```diff
@@ -15,10 +15,10 @@
 #include <hpx/iterator_support/traits/is_iterator.hpp>
 #include <hpx/lcos/dataflow.hpp>
 #include <hpx/lcos/future.hpp>
+#include <hpx/pack_traversal/unwrap.hpp>
```
Comment (hkaiser) [inline libs/segmented_algorithms/include/hpx/parallel/segmented_algorithms/detail/scan.hpp:18]: This module now depends on pack_traversal

File: libs/pack_traversal/src/pack_traversal.cpp (hunk 1 above)
Comment (msimberg) [reply libs/pack_traversal/src/pack_traversal.cpp:5]: ...

Well done me. Thanks @hkaiser!

File: libs/performance_counters/include/hpx/performance_counters/performance_counter_set.hpp (hunk 2 above)
Comment (msimberg) [reply libs/performance_counters/include/hpx/performance_counters/performance_counter_set.hpp:15]: This is another thing that we could check in CI (#4497).

**Comment by aurianer** [comment]
LGTM thanks!

**Comment by hkaiser** [comment]
@aurianer Looks like the documentation refers to one of the headers that were moved here.

**Comment by msimberg** [comment]
> @aurianer Looks like the documentation refers to one of the headers that were moved here.

@aurianer is innocent! It is the docs file list that's wrong, and I thought I fixed it... Will check again.

**Comment by msimberg** [comment]
All the CircleCI checks looked clean now. I've just rebased because of conflicts. Let's see if everything is still ok.

**Pull request merged**